core/domain/classifier_services.py

```diff
@@ -370,51 +468,53 @@ def get_classifier_from_exploration_attributes(exp_id, exp_version,
+        Exception: Entity for class ClassifierTrainingJob with id not found.
     """
-    classifier_exploration_mapping_model = (
-        classifier_models.ClassifierExplorationMappingModel.get_model(
+    training_job_exploration_mapping_model = (
+        classifier_models.TrainingJobExplorationMappingModel.get_model(
             exp_id, exp_version, state_name))
-    classifier_id = classifier_exploration_mapping_model.classifier_id
-    classifier = get_classifier_by_id(classifier_id)
-    return classifier
+    if training_job_exploration_mapping_model is None:
+        return None
+    job_id = training_job_exploration_mapping_model.job_id
+    classifier_training_job = get_classifier_training_job_by_id(job_id)
+    return classifier_training_job
 
 
-def create_classifier_exploration_mapping(exp_id, exp_version, state_name,
-                                          classifier_id):
-    """Creates an entry for Classifier Exploration Mapping in datastore.
+def create_job_exploration_mapping(exp_id, exp_version, state_name, job_id):
+    """Creates an entry for Training Job Exploration Mapping in datastore.
 
     Args:
         exp_id: str. ID of the exploration.
         exp_version: int. The exploration version.
         state_name: str. The state to which the classifier belongs.
-        classifier_id: str. ID of the classifier.
+        job_id: str. ID of the classifier training job.
 
     Returns:
-        str. ID of the classifier exploration mapping instance.
+        str. ID of the training job exploration mapping instance.
 
     Raises:
-        Exception: The Classifier-Exploration mapping with id already exists.
+        Exception: The TrainingJob-Exploration mapping with id already exists.
     """
-    classifier_exploration_mapping_model = (
-        classifier_models.ClassifierExplorationMappingModel.get_model(
+    training_job_exploration_mapping_model = (
+        classifier_models.TrainingJobExplorationMappingModel.get_model(
             exp_id, exp_version, state_name))
-    if classifier_exploration_mapping_model is not None:
-        raise Exception('The Classifier-Exploration mapping with id %s.%s.%s '
+    if training_job_exploration_mapping_model is not None:
+        raise Exception('The TrainingJob-Exploration mapping with id %s.%s.%s '
                         'already exists.' % (exp_id, exp_version,
                                              state_name.encode('utf-8')))
-    # Verify that the corresponding classifier exists.
-    get_classifier_by_id(classifier_id)
-
-    classifier_exploration_mapping = (
-        classifier_domain.ClassifierExplorationMapping(
-            exp_id, exp_version, state_name, classifier_id))
-    classifier_exploration_mapping.validate()
-    mapping_id = classifier_models.ClassifierExplorationMappingModel.create(
-        exp_id, exp_version, state_name, classifier_id)
+    # Verify that the corresponding job exists.
+    get_classifier_training_job_by_id(job_id)
```

Comment:
Done. This function doesn't exist anymore, but I did it for the create_multi version of this.